src/NHibernate/Hql/Ast/ANTLR/CollectionProperties.cs

```diff
@@ -11,34 +12,39 @@ namespace NHibernate.Hql.Ast.ANTLR
 	internal static class CollectionProperties
 	{
 
-		public static Dictionary<string, string> HQL_COLLECTION_PROPERTIES;
-
-		private static readonly string COLLECTION_INDEX_LOWER = CollectionPropertyNames.Index.ToLowerInvariant();
+		public static readonly Dictionary<string, string> HQL_COLLECTION_PROPERTIES = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
Usage of this collection looks like just a normal HashSet<string>(). As the class is internal, maybe it's better to replace it.